app/models/miq_cockpit_ws_worker/runner.rb

```diff
@@ -0,0 +1,196 @@
+class MiqCockpitWsWorker::Runner < MiqWorker::Runner
+  BINDING_ADDRESS = ENV['BINDING_ADDRESS'] || (Rails.env.production? ? nil : "0.0.0.0")
+
+  # Copied from old MiqProcess method
+  LINUX_STATES = {
+    'I' => :idle,             # Marks a process that is idle (sleeping for longer than about 20 seconds).
+    'R' => :runnable,         # Marks a runnable process.
+    'S' => :sleeping,         # Marks a process that is sleeping for less than about 20 seconds.
+    'T' => :stopped,          # Marks a stopped process.
+    'D' => :waiting,          # Marks a process in uninterruptible wait.  (Linux)
+    'Z' => :zombie            # Marks a dead process (a ``zombie'').
+  }.freeze
+
+  # Copied from old MiqProcess method
+  def find_pids(cmd)
+    pids = []
+
+    Dir['/proc/[0-9]*/cmdline'].each do |filename|
+      cmdline = MiqSystem.readfile_async(filename)
+      next if cmdline.nil?
+      matches = cmd =~ cmdline.tr("\000", " ").strip
+      next if matches.nil?
+      pid = filename.split('/')[2]
+      pids << pid.to_i
```

Comment:
Here's where that ps method is:  https://github.com/djberg96/sys-proctable/blob/2bc2fb9b42e8c1c8effd53ced186e1afb7c33428/lib/linux/sys/proctable.rb#L108

It returns a single struct if you pass a pid, otherwise, an array of structs.